modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/utils/JavaParserUtils.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.loader.parser.java.utils;
+
+import org.mule.runtime.api.meta.MuleVersion;
+import org.mule.runtime.extension.api.exception.IllegalModelDefinitionException;
+import org.mule.runtime.module.extension.api.loader.java.type.ExtensionParameter;
+import org.mule.runtime.module.extension.api.loader.java.type.FieldElement;
+import org.mule.runtime.module.extension.api.loader.java.type.MethodElement;
+import org.mule.runtime.module.extension.api.loader.java.type.Type;
+import org.mule.runtime.module.extension.api.loader.java.type.TypeGeneric;
+import org.mule.runtime.module.extension.api.loader.java.type.WithAnnotations;
+import org.mule.sdk.api.annotation.MinMuleVersion;
+
+import org.slf4j.Logger;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import static java.lang.String.format;
+import static java.util.Arrays.asList;
+import static java.util.Optional.of;
+import static org.mule.sdk.api.util.SdkApiConstants.NON_ENFORCE_MIN_MULE_VERSION_CLASSES;
+import static org.slf4j.LoggerFactory.getLogger;
+
+public class JavaParserUtils {
+
+  private static final Logger LOGGER = getLogger(JavaParserUtils.class);
+  private static final String firstMuleVersion = "4.1.0";
+
+  private static final Set<String> primitiveTypeNames =
+      new HashSet<>(asList("byte", "short", "int", "long", "float", "double", "char", "boolean", "void"));
+  private static final List<Class<?>> denyList = NON_ENFORCE_MIN_MULE_VERSION_CLASSES;
+
+  public static String calculateFromClass(Type clazz) {
```

Comment:
I don't think neither of these are calculating from, they're calculating **for**.